Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -214,6 +244,91 @@ noncomputable def powerBasisOfFiniteOfSeparable : PowerBasis F E :=
 
 end SeparableAssumption
 
+section FiniteIntermediateField
+
+theorem exists_primitive_element_of_finite_intermediateField
+    (h : Finite (IntermediateField F E)) : ∃ α : E, F⟮α⟯ = ⊤ := by
+  rcases finite_or_infinite F with (F_finite | F_inf)
+  · exact exists_primitive_element_of_finite_bot F E
+  · let P : IntermediateField F E → Prop := fun K ↦ ∃ α : E, F⟮α⟯ = K
+    have base : P ⊥ := ⟨0, adjoin_zero⟩
+    have ih : ∀ (K : IntermediateField F E) (x : E), P K → P (K⟮x⟯.restrictScalars F) := by
+      rintro K β ⟨α, rfl⟩
+      rw [adjoin_simple_adjoin_simple]
+      cases' primitive_element_inf_aux_of_finite_intermediateField F α β with γ hγ
+      exact ⟨γ, hγ.symm⟩
+    exact induction_on_adjoin P base ih ⊤
+
+-- TODO: move it to suitable file
+lemma _root_.IntermediateField.adjoin_eq_top_of_adjoin_eq_top (A B C: Type*)
+    [Field A] [Field B] [Field C] [Algebra A B] [Algebra B C] [Algebra A C] [IsScalarTower A B C]
+    {S : Set C} (hprim : adjoin A S = ⊤) : adjoin B S = ⊤ := by
+  apply restrictScalars_injective (K := A) (L' := B) (L := C)
+  rw [restrictScalars_top, ← top_le_iff, ← hprim, adjoin_le_iff,
+    coe_restrictScalars, ← adjoin_le_iff]
+
+-- TODO: give it a descriptive name if it's useful in other places
+private lemma finite_intermediateField_of_exists_primitive_element.aux_2
+    (α : E) (hprim : F⟮α⟯ = ⊤) (K K' : IntermediateField F E)
+    (heq : (minpoly K α).map (algebraMap K E) = (minpoly K' α).map (algebraMap K' E)) :
+    K ≤ K' := by
```

Comment:
This lemma should remain private, but I think the `K = K'` version (restated in terms of `Function.Injective`) is potentially useful outside of this file.

Reply:
Can you suggest a name for the `Function.Injective` version of the lemma?

Reply:
Actually, I think it's not harder to prove `K` is generated by the coefficients of `(minpoly K α).map (algebraMap K E)`, which would trivially imply injectivity, and would be of more value to mathlib.

Reply:
I think it's a good idea. It can be named `IntermediateField.eq_adjoin_minpoly_coeff_of_exists_primitive_element`?

Reply:
Sounds good to me!